File: sorting/bubble_sort.c
#include <stdio.h>
#include <stdlib.h>

// Sort array using bubble sort algorithm
void bubble_sort(int* array, size_t size){
    int tmp;
    for(int i = 0; i < size; i++){
        for(int j = 0; j < (size - i - 1); j++){
            if(array[j] > array[j+1]){
                // Swap values
                tmp = array[j];
                array[j] = array[j+1];
                array[j+1] = tmp;
            }
        }
    }
}

int main(){
    int array[] = {3, 2, -3, 9, 2, 0};
    for(int i = 0; i < 6; i++)
        printf("%d ", array[i]);

    bubble_sort(array, 6);
    printf("\nSorted:\n");
    for(int i = 0; i < 6; i++)
        printf("%d ", array[i]);
    printf("\n");
}

File: sorting/bubble_sort.py
from random import randint


def bubble_sort(array: list) -> None:
    """Sorts array using bubble sorting algorithm"""

    for i in range(len(array)):
        for j in range(len(array) - i - 1):
            if array[j] > array[j+1]:
                # Swap value of two elements
                array[j], array[j+1] = array[j+1], array[j]


if __name__ == "__main__":
    # Generate random array
    array = [randint(0, 9) for _ in range(20)]

    print("Array: ")
    for i in array:
        print("{} ".format(i), end='')

    bubble_sort(array)

    print("\nSorted array:")
    for i in array:
        print("{} ".format(i), end='')
    print("")
